convene-web/Gemfile

```diff
@@ -11,36 +11,61 @@ gem 'dotenv-rails', groups: [:development, :test]
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 6'
-# Use postgresql as the database for Active Record
-gem 'pg', '>= 0.18', '< 2.0'
+
+# Data Transport
+#
 # Use Puma as the app server
 gem 'puma', '~> 4.1'
+
+
+# Browser Layer
+#
 # Use SCSS for stylesheets
 gem 'sass-rails', '>= 6'
 # Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
 gem 'webpacker', '~> 5.0'
 # Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
 gem 'turbolinks', '~> 5'
+
+# API Layer
+#
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.7'
-# Use Redis adapter to run Action Cable in production
-# gem 'redis', '~> 4.0'
-
-# Use Active Storage variant
-# gem 'image_processing', '~> 1.2'
 
-# Reduces boot times through caching; required in config/boot.rb
-gem 'bootsnap', '>= 1.4.2', require: false
+# View Layer
+#
+# Allows us to render .markdown.erb files
+gem 'commonmarker'
 
-# Provides slug-based lookup for models
+# Database Layer
+#
+# Postgres extensions for ActiveRecord
+# @see https://github.com/GeorgeKaraszi/ActiveRecordExtended
+gem 'active_record_extended'
+# Slug-based model lookup
 gem 'friendly_id', '~> 5.2.4'
-# Use Active Model has_secure_password
+# Support Active Model has_secure_password
 gem 'bcrypt', '~> 3.1.7'
+# Use postgresql for data persistence
+gem 'pg', '>= 0.18', '< 2.0'
 
-
+# User-facing features
+#
+# Allows logging in via magic links
+gem 'passwordless'
 # Provides powerful administration features
 gem 'administrate'
 
+# Date/Time and Internationalization
+#
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+# Performance Optimization
+#
+# Reduces boot times through caching; required in config/boot.rb
+gem 'bootsnap', '>= 1.4.2', require: false
```

Comment:
Apologies for the noise here, I wanted to group the gems by their purpose. I probably should document that in this file.